src/logid/InputDevice.cpp

```diff
@@ -43,12 +43,7 @@ InputDevice::InputDevice(const char* name)
     device = libevdev_new();
     libevdev_set_name(device, name);
 
-    ///TODO: Is it really a good idea to enable all events?
     libevdev_enable_event_type(device, EV_KEY);
-    // KEY_ROTATE_LOCK_TOGGLE is the highest key mapped by
-    // /usr/share/X11/xkb/keycodes/evdev
-    for(unsigned int i = 0; i <= KEY_ROTATE_LOCK_TOGGLE; i++)
-        libevdev_enable_event_code(device, EV_KEY, i, nullptr);
```

Comment:
I"m a bit anxious about not enabling any key events at all. Systemd needs "ESC, numbers, and Q to D" (https://github.com/systemd/systemd/blob/c5b6b4b6d08cf4c16a871401358faeb5a186c02a/src/udev/udev-builtin-input_id.c#L324) to consider a device a "full keyboard". We might even enable A to Z and the common modifier keys by default to provide a baseline and avoid reclassifying our device from something else to a full keyboard at runtime.

Reply:
So one could (also for other implementations) check that the keys `0-127` (which includes the systemd keys) are enabled by default: 
https://github.com/jiixyj/libevdev/blob/d9089f7ed77feb963f0493299b4d56fe9c5b766a/include/linux/input-event-codes.h#L74-L203


And enable all other keys only on request. I think then one should be on the safe side, and no implementation should check that any keys above 127 are enabled.
And of course a "cache" for the keys would be good. I will work on that.